### PR TITLE
Implement more ec hints

### LINKED ISSUE #534: `EcRecoverDivModNPacked` hint (ec recover hint)
https://github.com/lambdaclass/cairo-vm/blob/db7fff18c9c1312024ebf4119fcdfeba23657b12/cairo_programs/ec_recover.cairo#L10

https://github.com/starkware-libs/cairo-lang/blob/0e4dab8a6065d80d1c726394f5d9d23cb451706a/src/starkware/starknet/security/whitelists/ec_recover.json#L4

https://github.com/lambdaclass/cairo-vm/blob/db7fff18c9c1312024ebf4119fcdfeba23657b12/vm/src/hint_processor/builtin_hint_processor/hint_code.rs#L1325C11-L1325C38

### LINKED ISSUE #592: EcRecoverSubAB hint
https://github.com/lambdaclass/cairo-vm/blob/db7fff18c9c1312024ebf4119fcdfeba23657b12/cairo_programs/ec_recover.cairo#L31

https://github.com/starkware-libs/cairo-lang/blob/0e4dab8a6065d80d1c726394f5d9d23cb451706a/src/starkware/starknet/security/whitelists/ec_recover.json#L15

https://github.com/lambdaclass/cairo-vm/blob/db7fff18c9c1312024ebf4119fcdfeba23657b12/vm/src/hint_processor/builtin_hint_processor/hint_code.rs#L1364

### LINKED ISSUE #593: EcRecoverProductMod hint
https://github.com/lambdaclass/cairo-vm/blob/db7fff18c9c1312024ebf4119fcdfeba23657b12/cairo_programs/ec_recover.cairo#L53

https://github.com/starkware-libs/cairo-lang/blob/0e4dab8a6065d80d1c726394f5d9d23cb451706a/src/starkware/starknet/security/whitelists/ec_recover.json#L27

https://github.com/lambdaclass/cairo-vm/blob/db7fff18c9c1312024ebf4119fcdfeba23657b12/vm/src/hint_processor/builtin_hint_processor/hint_code.rs#L1375

### LINKED ISSUE #594: EcRecoverProductDivM hint
https://github.com/lambdaclass/cairo-vm/blob/db7fff18c9c1312024ebf4119fcdfeba23657b12/cairo_programs/ec_recover.cairo#L68

https://github.com/starkware-libs/cairo-lang/blob/0e4dab8a6065d80d1c726394f5d9d23cb451706a/src/starkware/starknet/security/whitelists/ec_recover.json#L41

https://github.com/lambdaclass/cairo-vm/blob/db7fff18c9c1312024ebf4119fcdfeba23657b12/vm/src/hint_processor/builtin_hint_processor/hint_code.rs#L1407

### PR DESCRIPTION
**Pull request by har777**
Fixes #534 
Fixes #592 
Fixes #593 
Fixes #594

### CODE REVIEW

File: pkg/hintrunner/zero/zerohint_ec.go [hunk 1]
```diff
@@ -1094,3 +1094,299 @@ func createRandomEcPointHinter(resolver hintReferenceResolver) (hinter.Hinter, e
...
+			//> value = res = div_mod(x, s, N)
+			resBig, err := secp_utils.Divmod(&xPackedBig, &sPackedBig, &nPackedBig)
+			if err != nil {
+				return err
+			}
+
+			valueBig := new(big.Int)
+			valueBig.Set(&resBig)
+
+			return ctx.ScopeManager.AssignVariables(map[string]any{"res": &resBig, "value": valueBig})
```
Comment (TAdev0) [inline pkg/hintrunner/zero/zerohint_ec.go:1169]: i think we only need `res` in scope here, at least given the integration test. But there is no cairo file using this hint in cairolang repo, so its probably more cautious to add both `res` and `value` in scope

File: pkg/hintrunner/zero/zerohint_ec.go [hunk 2]
```diff
@@ -1094,3 +1094,299 @@ func createRandomEcPointHinter(resolver hintReferenceResolver) (hinter.Hinter, e
...
+	}
+
+	s, err := resolver.GetResOperander("s")
+	if err != nil {
+		return nil, err
+	}
+
+	return newEcRecoverDivModNPackedHint(n, x, s), nil
+}
+
```
Comment (TAdev0) [inline pkg/hintrunner/zero/zerohint_ec.go:1192]: missing comment of the hint

File: pkg/hintrunner/zero/zerohint_ec.go [hunk 3]
```diff
@@ -1094,3 +1094,299 @@ func createRandomEcPointHinter(resolver hintReferenceResolver) (hinter.Hinter, e
...
+			resBig := new(big.Int)
+			resBig.Sub(&aPackedBig, &bPackedBig)
+			if err != nil {
+				return err
+			}
+
+			valueBig := new(big.Int)
+			valueBig.Set(resBig)
+
+			return ctx.ScopeManager.AssignVariables(map[string]any{"res": resBig, "value": valueBig})
```
Comment (TAdev0) [inline pkg/hintrunner/zero/zerohint_ec.go:1247]: same but probably to keep both

File: pkg/hintrunner/zero/zerohint_ec.go [hunk 4]
```diff
@@ -1094,3 +1094,299 @@ func createRandomEcPointHinter(resolver hintReferenceResolver) (hinter.Hinter, e
...
+	}
+
+	b, err := resolver.GetResOperander("b")
+	if err != nil {
+		return nil, err
+	}
+
+	return newEcRecoverSubABHint(a, b), nil
+}
+
```
Comment (TAdev0) [inline pkg/hintrunner/zero/zerohint_ec.go:1265]: missing hint comment

File: pkg/hintrunner/zero/zerohint_ec.go [hunk 5]
```diff
@@ -1094,3 +1094,299 @@ func createRandomEcPointHinter(resolver hintReferenceResolver) (hinter.Hinter, e
...
+	}
+
+	m, err := resolver.GetResOperander("m")
+	if err != nil {
+		return nil, err
+	}
+
+	return newEcRecoverProductModHint(a, b, m), nil
+}
+
```
Comment (TAdev0) [inline pkg/hintrunner/zero/zerohint_ec.go:1362]: missing hint comment

**Review by TAdev0 — COMMENTED** [review]
left minor comments otherwise lgtm

**Comment by MaksymMalicki** [comment]
Looks good to me! Outside of the hints annotations, nothing to be added. Approve

**Review by MaksymMalicki — APPROVED** [review]

**Pull request merged**